src/InMemoryMessageRepository.php

```diff
@@ -44,12 +44,15 @@ public function persist(Message ...$messages)
 
     public function retrieveAll(AggregateRootId $id): Generator
```

Comment:
Matter of opinion, but I think this is cleaner.
However not sure about performance on this one.

```suggestion
    public function retrieveAll(AggregateRootId $id): Generator
    {
        $messages = array_filter(
            $this->messages,
            function (Message $message) use ($id): bool  {
                return $id->toString() === $message->header(Header::AGGREGATE_ROOT_ID)->toString();
            }
        );
    
        yield from $messages;
        
        if (null === $lastKey = array_key_last($messages)) {
        	return 0;
        }
    
        return $messages[$lastKey]->aggregateVersion();
    }
```

Reply:
It looks more fancy, but there's a lot more going on. With extra function calls and such, I prefer the stupid way I got in the PR. But I appreciate the feedback :)